python/revng/cli/_commands/fetch_debuginfo/elf.py

```diff
@@ -0,0 +1,195 @@
+                # TODO: Shell we check the `checksum` to validate the file?
+                return absolute_path_debuginfo
+
+            log(f"Did not find {debug_file_name} in the current directory...", verbose)
+            log("Looking into .debug/...", verbose)
+
+            # 2B) Look into ${directory_of_executable}/.debug/.
+            absolute_path_debuginfo = os.path.join(dir_from_exe, ".debug/", debug_file_name)
+            if os.path.exists(absolute_path_debuginfo):
+                log("Debug File found: " + absolute_path_debuginfo, verbose)
+                return absolute_path_debuginfo
+
+            log(f"Did not find {debug_file_name} in .debug/ directory...", verbose)
+            log("Looking into /usr/lib/debug/...", verbose)
+            # 2C) Look into /usr/lib/debug/, (e.g. /usr/lib/debug/usr/bin/ls.debug).
+            if os.path.exists(global_debug_directory):
+                absolute_path_debuginfo = global_debug_directory + dir_from_exe
+                absolute_path_debuginfo = os.path.join(absolute_path_debuginfo, debug_file_name)
+                if os.path.exists(absolute_path_debuginfo):
+                    log("Debug File found: " + absolute_path_debuginfo, verbose)
+                    return absolute_path_debuginfo
+
+            log(f"Could not find {debug_file_name}...", verbose)
+            # TODO: Should we `return None` here?
+        else:
+            log("error: Unable to parse of debug info file from .gnu_debuglink", verbose)
+            # TODO: Should we `return None` here?
+
+    build_id_dir = global_debug_directory + ".build-id/"
+    if not os.path.exists(build_id_dir):
+        log(f"Could not find {build_id_dir}", verbose)
+        return None
+
+    # 3) Build ID.
+    #  If build-id is `abcdef1234`, we look for:
+    #   - /usr/lib/debug/.build-id/ab/cdef1234.debug
+    log("Parsing build-id...", verbose)
+    biuld_id = parse_build_id(file)
+    if biuld_id is None:
+        # If we cannot parse the build id, we cannot fetch debug info.
+        return None
+
+    log("BUILD-ID: " + biuld_id, verbose)
+    # NOTE: The first two bytes form a directory inside `/usr/lib/debug/.build-id/` and
+    # rest of the hash is used for name of the debug info file.
+    first_two_chars = biuld_id[:2]
+    the_rest_of_build_hash = biuld_id[2:]
+    debug_file_name = the_rest_of_build_hash + ".debug"
+    absolute_path_debuginfo = os.path.join(build_id_dir, first_two_chars, debug_file_name)
+    if os.path.exists(absolute_path_debuginfo):
+        log("Debug File found: " + absolute_path_debuginfo, verbose)
+        return absolute_path_debuginfo
+    else:
+        log(f"Could not find {absolute_path_debuginfo}", verbose)
+
+    # 4) Find debug info on the web as `debuginfod` does.
+    if "XDG_CACHE_HOME" not in os.environ or os.environ["XDG_CACHE_HOME"] == "":
+        log("warning: XDG_CACHE_HOME was not set. Using ~/.cache/debuginfod_client/...", verbose)
```

Comment:
Consider making a `warning` function that prefixes `Warning:`. One day we might use a colorful logger and will make things easier.